Libraries/LibMarkdown/List.cpp

```diff
@@ -66,21 +66,24 @@ String List::render_for_terminal() const
     return builder.build();
 }
 
-bool List::parse(Vector<StringView>::ConstIterator& lines)
+OwnPtr<List> List::parse(Vector<StringView>::ConstIterator& lines)
 {
+    auto list = make<List>();
     bool first = true;
     size_t offset = 0;
     StringBuilder item_builder;
     auto flush_item_if_needed = [&] {
         if (first)
-            return;
+            return true;
 
-        Text text;
-        bool success = text.parse(item_builder.string_view());
-        ASSERT(success);
-        m_items.append(move(text));
+        auto text = Text::parse(item_builder.string_view());
+        if (!text.has_value())
+            return false;
+
+        list->m_items.append(move(text.value()));
```

Comment:
Could we, perhaps, instead of appending items directly to `list->m_items`, keep our own vector of items, and pass it to the `List` constructor, along with whether it's ordered, at the very end?